### PR TITLE
Added support for SOCKS5 proxy

### PR DESCRIPTION
**Pull request by carmebar**
Now you're able to use a SOCKS5 proxy as stated in the TODO file.
This will allow to route traffic via Tor (default SOCKS5 proxy-> 127.0.0.1:9050).
This will also allow to reach hidden service remove calendars because it routes DNS queries via the SOCKS5 proxy.

And thanks for the great project :smile:

### CODE REVIEW

File: otsclient/cmds.py [hunk 1]
```diff
@@ -110,6 +139,8 @@ def submit_async_thread(remote, msg, q):
 
 
 def stamp_command(args):
+    install_socks5(args.socks5_proxy)
```
Comment (petertodd) [inline otsclient/cmds.py]: `install_socks()` should be elsewhere; stamp is only one of many commands that uses remote calendars. You probably should put that where arguments are parsed instead in `otsclient/args.py` in the `parse_ots_args()` function.

File: otsclient/cmds.py [hunk 2]
```diff
@@ -38,6 +49,24 @@
 
 import otsclient
 
+def install_socks5(proxy_url):
+    """Monkey patching socket library to redirect via SOCKS5"""
```
Comment (petertodd) [inline otsclient/cmds.py]: That should be "Monkey patch socket library to redirect via SOCKS4"

Comment (carmebar) [inline otsclient/cmds.py]: It's SOCKS5. But yeah, I'll fix the typo on "patching".

**Comment by carmebar** [comment]
Thanks for the insight, I'll fix the pull request tonight.

File: otsclient/cmds.py (hunk 2 above)
Comment (petertodd) [inline otsclient/cmds.py]: Lol, yeah, that's a typo on my part too!

**Comment by carmebar** [comment]
Much cleaner version now. Hopefully ready to merge :grin:

**Comment by petertodd** [comment]
Thanks! I added three more commits to allow the PySocks dependency to be optional, and fix some minor nits: 

https://github.com/opentimestamps/opentimestamps-client/commit/2c1cd68f32b0ea6ee9f9f25b7a34f73c60076ed3

https://github.com/opentimestamps/opentimestamps-client/commit/6231a8bd7411f45900c9423b19aac9fe8de6951b

https://github.com/opentimestamps/opentimestamps-client/commit/2cadcd20cb77450ee227707a2909caa9d21f52df

**Pull request merged**